src/workerd/server/alarm-scheduler.c++

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "alarm-scheduler.h"
+
+namespace workerd::server {
+
+int AlarmScheduler::maxJitterMsForDelay(kj::Duration delay) {
+  double delayMs = delay / kj::MILLISECONDS;
+  return std::floor(RETRY_JITTER_FACTOR * delayMs);
+}
+
+std::default_random_engine makeSeededRandomEngine() {
+  // Using the time as a seed here is fine, we just want to have some randomness for retry jitter
+  auto time = kj::systemPreciseMonotonicClock().now();
+  auto seed = (time - kj::origin<kj::TimePoint>()) / kj::NANOSECONDS;
+
+  std::default_random_engine engine(seed);
+  return engine;
+}
+
+AlarmScheduler::AlarmScheduler(
+    const kj::Clock& clock,
+    kj::Timer& timer,
+    const SqliteDatabase::Vfs& vfs,
+    kj::PathPtr path)
+    : clock(clock), timer(timer), random(makeSeededRandomEngine()),
+      db([&]{
+        auto db = kj::heap<SqliteDatabase>(vfs, path,
+            kj::WriteMode::CREATE | kj::WriteMode::MODIFY | kj::WriteMode::CREATE_PARENT);
+        ensureInitialized(*db);
+        return kj::mv(db);
+      }()),
+      tasks(*this) {
+    auto now = clock.now();
+
+    // TODO(someday): don't maintain the entire alarm set in memory -- right now for the usecase of
+    // local development, doing so is sufficent.
+    auto query = db->run(R"(
```

Comment:
nit: Can I convince you to put this in a named function that you call here? Hopefully it should make it easier for another dev to quickly understand what this query is about.